src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -56,15 +58,45 @@ Copyright (C) 2016 Xamarin. All rights reserved.
   <Message Text="IntermediateOutputPath: $(IntermediateOutputPath)" />
 </Target>
 
+<Target Name="_GetProjectResourceDirectory" Returns="@(_ProjectReferenceResourceDirectory)">
+  <ItemGroup>
+    <_ProjectReferenceResourceDirectory Include="$(MSBuildProjectDirectory)\$(MonoAndroidResourcePrefix)"
+      StampFile="$(MSBuildProjectFile)"
+      Condition="!Exists('$(OutputPath)$(TargetName).aar')"
+    />
+  </ItemGroup>
+</Target>
+
+<Target Name="_CollectProjectReferenceResources"
+    Condition="'$(AndroidUseDesignerAssembly)' == 'True' And '$(DesignTimeBuild)' == 'True' "
+>
+  <MSBuild
+      Projects="@(ProjectReference)"
+      Targets="_GetProjectResourceDirectory"
+      SkipNonexistentTargets="true"
+  >
+    <Output TaskParameter="TargetOutputs" ItemName="LibraryResourceDirectories" />
+  </MSBuild>
+</Target>
+
+<Target Name="_CalculateDesignTimeAars" Condition=" '$(DesignTimeBuild)' == 'True' ">
+  <ItemGroup>
+    <!-- Only use the aar files if we have not extracted the data -->
+    <_DesignTimeAarFiles Include="@(AndroidAarLibrary)" Condition=" '@(LibraryResourceDirectories->Count())' == '0' " />
+    <_DesignTimeAarFiles Include="@(LibraryProjectZip)" Condition="'%(LibraryProjectZip.Extension)' == '.aar' and '@(LibraryResourceDirectories->Count())' == '0'" />
```

Comment:
This is probably not worth changing, but I wonder if it's better if this expression was first:
```
'@(LibraryResourceDirectories->Count())' == '0'
```

So, it would prevent MSBuild from batching/expanding the first part:
```
'%(LibraryProjectZip.Extension)' == '.aar'
```

It probably runs the `->Count()` N times, but probably not slow enough to matter, though.